quaddtype/quaddtype/src/dtype.c

```diff
@@ -1,197 +1,214 @@
 #include <Python.h>
+#include <sleef.h>
+#include <sleefquad.h>
 
-#define PY_ARRAY_UNIQUE_SYMBOL quaddtype_ARRAY_API
-#define PY_UFUNC_UNIQUE_SYMBOL quaddtype_UFUNC_API
+#define PY_ARRAY_UNIQUE_SYMBOL QuadPrecType_ARRAY_API
+#define PY_UFUNC_UNIQUE_SYMBOL QuadPrecType_UFUNC_API
 #define NPY_NO_DEPRECATED_API NPY_2_0_API_VERSION
 #define NPY_TARGET_VERSION NPY_2_0_API_VERSION
 #define NO_IMPORT_ARRAY
 #define NO_IMPORT_UFUNC
-#include "numpy/ndarraytypes.h"
 #include "numpy/arrayobject.h"
+#include "numpy/ndarraytypes.h"
 #include "numpy/dtype_api.h"
+#include "numpy/_public_dtype_api_table.h" // not included in dtype_api.h
```

Comment:
`PyArrayDTypeMeta_Type` is defined inside the `_public_dtype_api_table.h` for setting the `ob_type` as
```
    ((PyObject *)&QuadPrecDType)->ob_type = &PyArrayDTypeMeta_Type;
```

Reply:
Ah, you should be doing `#include "numpy/ndarraytypes.h"` instead of including the private header. It's a core data type of NumPy so it can't live in the dtype API header.